File: DataStructures/Tries/Node.cpp
#include "Node.h"
Node::Node() {

}
Node::~Node() {
	std::map<char, Node*>::iterator iter = this->letters.begin();
	while (iter != this->letters.end()) {
		if (iter->second != nullptr) {
			delete iter->second;
		}
		iter->second = nullptr;
		iter++;
	}
	
}
void Node::Insert(const char word[],const int count,int i) {
	if (i >= count) {
		this->letters['*'] = nullptr;
		return;
	}
	const char letter = word[i];
	if (this->letters.count(letter) == 0) {
		this->letters[letter] = new Node();
	}
	this->letters[letter]->Insert(word,count,++i);
}
bool Node::Contains(const char word[],const int count,int i) {
	if (i >= count) return true;
	const char letter = word[i];
	if (this->letters.count(letter) > 0) {
		return this->letters[letter]->Contains(word,count,++i);
	}
	return false;
}

File: src/tree/Trie/node.ts
export default class Node{
    private letters:Map<string,Node|null>;
    constructor() {
        this.letters = new Map();
    }
    insert(word:string,index:number):void{
        if(index >= word.length){
            this.letters.set('*',null);
            return;
        }
        const letter:string = word[index];
        if(!this.letters.has(letter)){
            this.letters.set(letter,new Node());
        }
        // @ts-ignore
        this.letters.get(letter).insert(word,++index);
    }
    contains(word:string,index:number):boolean{
        if(index >= word.length) return true;
        const letter = word[index];
        if(this.letters.has(letter)){
            // @ts-ignore
            return this.letters.get(letter).contains(word,++index);
        }
        return false;
    }
}
